frontend/packages/console-shared/src/hooks/perspectiveUtils.ts

```diff
@@ -1,7 +1,125 @@
+  state: PerspectiveVisibilityState;
+  accessReview?: PerspectiveAccessReview;
+};
+
+type Perspective = {
+  id: string;
+  visibility: PerspectiveVisibility;
+};
+
 export const getPerspectiveVisitedKey = (perspective: PerspectiveType): string =>
   `${USERSETTINGS_PREFIX}.${PERSPECTIVE_VISITED_FEATURE_KEY}.${perspective}`;
+
+export const hasReviewAccess = async (accessReview: PerspectiveAccessReview) => {
+  let hasAccess: boolean = true;
+  const requiredAccessReview = accessReview?.required;
+  const missingAccessReview = accessReview?.missing;
+  const requiredPromises = requiredAccessReview?.map((resourceAttributes) =>
+    checkAccess(resourceAttributes),
+  );
+  const missingPromises = missingAccessReview?.map((resourceAttributes) =>
+    checkAccess(resourceAttributes),
+  );
+
+  requiredPromises?.length > 0 &&
+    (await Promise.all(requiredPromises)
+      .then((values) => {
+        // enable the prespective if all the access review checks are successful
+        hasAccess = hasAccess && values.every((val) => val.status.allowed);
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        console.warn('AccessReview check failed', e);
+        hasAccess = hasAccess && false;
+      }));
+
+  missingPromises?.length > 0 &&
+    (await Promise.all(missingPromises)
+      .then((values) => {
+        // enable perspective if atleast one access review check fails
+        hasAccess = hasAccess && !values.every((val) => val.status.allowed);
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        console.warn('AccessReview check failed', e);
+        hasAccess = hasAccess && true;
+      }));
+
+  return hasAccess;
+};
+
+export const useGetPerspectives = (): LoadedExtension<PerspectiveExtension>[] => {
+  const perspectiveExtensions = useExtensions<PerspectiveExtension>(isPerspective);
+  const [extensions, setExtensions] = React.useState<LoadedExtension<PerspectiveExtension>[]>([]);
+  const adminPerspective = perspectiveExtensions.find(
+    (perspectiveEx) => perspectiveEx.properties.id === 'admin',
+  );
+
+  const setResults = React.useCallback(
+    (newResult: LoadedExtension<PerspectiveExtension>) => {
+      setExtensions((oldResults: LoadedExtension<PerspectiveExtension>[]) => {
+        const arr = [...oldResults, newResult];
+        return _.uniqBy(arr, 'properties.id');
+      });
+    },
+    [setExtensions],
+  );
+  React.useEffect(() => {
+    if (!window.SERVER_FLAGS.perspectives) {
+      setExtensions(perspectiveExtensions);
+    } else {
+      const perspectives: Perspective[] = JSON.parse(window.SERVER_FLAGS.perspectives);
+      perspectiveExtensions.forEach((perspectiveExtension) => {
+        const perspective = perspectives?.find((p) => p.id === perspectiveExtension.properties.id);
+
+        if (perspective?.visibility?.state === PerspectiveVisibilityState.Enabled) {
+          setResults(perspectiveExtension);
+        } else if (
```

Comment:
Unconfigured perspectives should also be shown?!

```suggestion
        if (!perspective || !perspective.visibility || perspective.visibility.state === PerspectiveVisibilityState.Enabled) {
          setResults(perspectiveExtension);
        } else if (
```

Or you check for Disabled and add everything else in the else case..